tests/integration/bundle-testing.yaml.j2

```diff
@@ -0,0 +1,55 @@
+---
```

Comment:
I maintain the [juju-bundle](https://snapcraft.io/juju-bundle) plugin, which is designed to handle this sort of bundle management, instead of having a separate bundle template that can fall out of sync with the main `bundle.yaml`. I'd be interested in hearing more about how you're using this template. I assume you're using it for local development in addition to the CI job here, and building each charm manually?

If you were to create a `charms` directory at the root of this repo and create a symlink inside for each charm with e.g. `ln -s ../../prometheus-operator prometheus`, you could run `juju bundle deploy --build` to have each charm built and deployed as part of the bundle. Does that meet your needs vs a bundle template? Otherwise, I'd be interesting in see if `juju-bundle` could be expanded to meet any extra needs.

Reply:
I am not aware of a `--build` arg to any juju command. Could you please elaborate?
A few commits back I was using tox commands to `git clone` and/or `charmcraft build` but I think now it is cleaner.

`bundle-testing.yaml.j2` has a few use cases:
- when rendered with defaults, everything comes from charmhub. This is useful for tests intended for the bundle itself.
- CI workflow of the individual charms: each charm has their own integration tests, but also runs the bundle's tests, but with one modification: the charm under test is built locally but everything else comes from charmhub.
- local testing when making breaking changes would include re-running the bundle tests, but with a number of local charms .

The template approach works well with all these use cases.

Reply:
I like the template approach but I think it should definitely not be separate from the production bundle because they could get out of sync. I think the best approach would be to have a bundle.yaml.j2 at the root of the tree and a makefile or script to generate the bundle.

Reply:
The `--build` arg is provided by the [juju-bundle](https://snapcraft.io/juju-bundle) plugin, not directly by any core juju command.

I will add in a flag such as `--build-app=foo` / `--build-app=foo=../../foo-operator` to the plugin that will allow fine-grained control for which apps to build and optionally from where, which I think takes care of each of the use cases. That isn't something that should block this PR or anything, but let me know if that seems like something that wouldn't work for you in general.

Reply:
captured in #8